tempodb/tempodb.go

```diff
@@ -466,6 +478,36 @@ func (rw *readerWriter) updateBlocklist(tenantID string, add []*backend.BlockMet
 	rw.compactedBlockLists[tenantID] = append(rw.compactedBlockLists[tenantID], compactedAdd...)
 }
 
+func (rw *readerWriter) shouldCache(meta *backend.BlockMeta, curTime time.Time) bool {
+	// compaction level is _atleast_ CacheMinCompactionLevel
+	if rw.cfg.CacheMinCompactionLevel > 0 && meta.CompactionLevel < rw.cfg.CacheMinCompactionLevel {
+		return false
+	}
+
+	// block is not older than CacheMaxBlockAge
+	if rw.cfg.CacheMaxBlockAge > 0 && curTime.Sub(meta.StartTime) > rw.cfg.CacheMaxBlockAge {
```

Comment:
⚠️  _This comment is about the declaration of `BlockMeta.StartTime`, but I can't add a comment there in GitHub._

The field declaration of BlockMeta.StartTime has a comment 
```
// Currently mostly meaningless but roughly matches to the time the first obj was written to this block
```
Can we update this? Since we use this to make caching decisions it's not meaningless anymore.